Modules/Filtering/MathematicalMorphology/test/itkTopHatImageFilterTest.cxx

```diff
@@ -24,14 +24,68 @@
 #include "itkImageFileWriter.h"
 #include "itkTestingMacros.h"
 
+
+template <typename TKernelImageFilter>
+int
+itkTopHatImageFilterTestHelper(TKernelImageFilter *                    filter,
+                               typename TKernelImageFilter::KernelType ball,
+                               const bool                              safeBorder,
+                               const int                               algorithm,
+                               const bool                              forceAlgorithm,
+                               const char *                            inputFileName,
+                               const char *                            outputFileName)
+{
+  // Declare the reader and writer
+  using ReaderType = itk::ImageFileReader<TKernelImageFilter::InputImageType>;
+  ReaderType::Pointer reader = ReaderType::New();
```

Comment:
`ReaderType::Pointer` needs to be `typename ReaderType::Pointer`

Reply:
Fixed in 18d1de6. Thanks for spotting these !